Source/ProgressionSystemRuntime/Private/Components/PSSpotComponent.cpp

```diff
@@ -49,8 +49,21 @@ void UPSSpotComponent::BeginPlay()
 // Clears all transient data created by this component.
 void UPSSpotComponent::OnUnregister()
 {
+	// reset back to initial state. By default, the spot is unlocked 
+	constexpr bool bSpotUnlocked = true;
+	if (PlayerSpotOnLevelInternal)
+	{
+		PlayerSpotOnLevelInternal->SetActive(bSpotUnlocked);
+		PlayerSpotOnLevelInternal = nullptr;
+	}
+
+	if (SaveGameInstanceInternal)
```

Comment:
Save is already destroyed in PerformCleanup of the subsystem function.

Regarding Spot component itself, please remove this `SaveGameInstanceInternal` cached variable from own header file, it shouldn't be cached and cleaned up by Component, its responsibility only of the subsystem

Reply:
done with: https://github.com/h4rdmol/ProgressionSystem/pull/26/commits/09bfccd83e624edaaea9d505ebdd5e754500b050

Reply:
done with https://github.com/h4rdmol/ProgressionSystem/pull/26/commits/97e76a5619eded91ce0aa9d379102db1a0a5ed72